File: sosu-frontend/src/services/FunctionalStateService.ts
import http from "./http.client"
import type {FunctionalState} from "@/models/FunctionalState";

export class FunctionalStateService {
    async getFunctionalStateByCitizen(citizenId: string): Promise<FunctionalState[]>{
        const res = await http.get<FunctionalState[]>("/api/FunctionalState/"+citizenId);
        return res.data;
    }
}

File: SOSU2022_BackEnd/SOSU2022_BacEnd.Domain/Services/FunctionalStateService.cs
using System.Collections.Generic;
using System.IO;
using SOSU2022_BacEnd.Domain.IRepositories;
using SOSU2022_BackEnd.Core.IServices;
using SOSU2022_BackEnd.Core.Models;


namespace SOSU2022_BacEnd.Domain.Services
{
    public class FunctionalStateService : IFunctionalStateService
    {
        private readonly IFunctionalStateRepository _repo;

        public FunctionalStateService(IFunctionalStateRepository repo)
        {
            _repo = repo ?? throw new InvalidDataException();
        }
        
        public List<FunctionalState> GetByCitizen(string citizenId)
        {
            return _repo.GetByCitizen(citizenId);
        }
    }
}